### PR TITLE
Architecture doc update

### PR DESCRIPTION
**Pull request by joshjung**
## Hey, I just made a Pull Request!

Some updates primarily to the architecture overview page to make things clearer, updating some visuals, fixing some broken links, and updating the documentation a bit in the microsite readme.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by joshjung** [comment]
FYI I have no clue why the above is failing! It appears to be a fluke, @Rugvip is there a way to trigger a rerun of the pipeline without committing something?

**Comment by joshjung** [comment]
Bump on this! Not sure how to fix the build

**Comment by aramissennyeydd** [comment]
@joshjung I think the comment doesn't get deleted if the DCO check passes after it fails initially, you can see those results [here](https://github.com/backstage/backstage/pull/24690/checks?check_run_id=24739702254). It should be safe to ignore the comment above.

**Review by freben — COMMENTED** [review]
Hai fren! Long time no see.

File: docs/overview/architecture-overview.md [hunk 1]
```diff
@@ -6,149 +6,133 @@ description: Documentation on Architecture overview
...
+in the `plugins` folder. For example, the source code for the catalog plugin is available at [plugins/catalog](https://github.com/backstage/backstage/tree/master/plugins/catalog).
 
-### Installing plugins
+### Installing Plugins
 
-Plugins are typically installed as React components in your Backstage
-application. For example,
-[here](https://github.com/backstage/backstage/blob/master/packages/app/src/App.tsx)
-is a file that imports many full-page plugins in the Backstage sample app.
+Plugins are typically installed as React components built and co-located in the same repository as your
```
[the remark this answers is not in the document]
Comment (joshjung) [reply docs/overview/architecture-overview.md:69]: Okay interesting. So you know, when I was trying to explain how Backstage works to people at work a month ago, this was the page I was wanting to send to them, and I realized it didn't really "sell" the possibilities that it had with the plugin system quite as well as it was working in 2024, so was hoping to update it to make it clear that this page on the architecture would provide that full high-level view of every angle an org could take in using Backstage, but perhaps I missed the original intent of this page.

File: docs/overview/architecture-overview.md [hunk 2]
```diff
@@ -6,149 +6,133 @@ description: Documentation on Architecture overview
 
 ## Terminology
 
-Backstage is constructed out of three parts. We separate Backstage in this way
-because we see three groups of contributors that work with Backstage in three
-different ways.
+Early on, we envisioned three primary types of contributors to a Backstage installation: core, application, and plugin
```
[the remark this answers is not in the document]
Comment (joshjung) [reply docs/overview/architecture-overview.md:9]: Okay so do you see this architecture doc as the architecture related *specifically* to an orgs installation?

[the remark this answers is not in the document]
Comment (freben) [reply docs/overview/architecture-overview.md:9]: Hm at least I guess it's explaining all of the concepts with a focus on what your experience will be as an integrator/adopter. But mainly my observation was that your text reads to me as "your installation will have core contributors", and I kind of thought that the core contributors are only relevant in the open source repo.

**Review by stevenmchaves — APPROVED** [review]

**Comment by joshjung** [comment]
Yeah I need to clean up this PR! Hopefully will have time sometime this weekend.

**Pull request closed without merging**